File: java-hfsframework-primefaces/hfsframework-admin/src/main/java/br/com/hfsframework/admin/data/AdmFuncionarioRepository.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.admin.data;

import java.math.BigDecimal;

import org.apache.deltaspike.data.api.EntityRepository;
import org.apache.deltaspike.data.api.Query;
import org.apache.deltaspike.data.api.Repository;

import br.com.hfsframework.admin.model.AdmFuncionario;

/**
 * The Interface AdmFuncionarioRepository.
 */
@Repository(forEntity = AdmFuncionario.class)
public interface AdmFuncionarioRepository extends EntityRepository<AdmFuncionario, Long> {

	/**
	 * Find by matricula and CPF.
	 *
	 * @param id
	 *            the id
	 * @param cpf
	 *            the cpf
	 * @return the vw adm funcionario
	 */
	@Query(named = "AdmFuncionario.findByMatriculaAndCPF")
	AdmFuncionario findByMatriculaAndCPF(Long id, BigDecimal cpf);
}

File: csharp-hfsframework/hfsframework-admin/hfsframework.admin.data/AdmFuncionarioRepository.cs
using hfsframework.hfsbase;
using hfsframework.admin.model;
using hfsframework.util;
using NHibernate;

namespace hfsframework.admin.data
{
    public class AdmFuncionarioRepository: EntityRepository<AdmFuncionario, long>
    {
        public AdmFuncionario findByMatriculaAndCPF(long id, long cpf)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionario.findByMatriculaAndCPF");
                q.SetInt64(0, id);
                q.SetInt64(1, cpf);
                return q.UniqueResult<AdmFuncionario>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

    }
}
